### PR TITLE
Elastic force to prevent accidental dismiss of bottom sheet & Allow bottomsheet to open as expanded when height smaller than parent's.

### PR DESCRIPTION
**Pull request by sergio-moura88**
Elastic force to prevent accidental dismiss of bottom sheet.
If we move just a pixel, the bottom sheet is dismissed. with this feature, its behaviour is more like the native android bottomsheet's.

Allow bottomsheet to open as expanded when height smaller than parent's.
When you have a bottomsheet which height is smaller than the BottomSheetLayout's height, the bottomsheet starts (like it should when you call show). If the bottomsheet view is a scrollview, the first time you try to scroll, the bottomsheet doesn't allow because try to make it expanded, and only after that you can scroll. 
With this fix, if, for instance you put the bottomsheet view with 300dp height, and it is a listview, it just starts with Expanded state, and so you can always scroll inside that.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.0.0-alpha3'
```
Comment (hzsweers) [inline build.gradle]: Please revert this line

File: bottomsheet/src/main/java/com/flipboard/bottomsheet/BottomSheetLayout.java [hunk 2]
```diff
@@ -101,24 +101,35 @@ public void transformView(float translation, float maxTranslation, float peekedT
     private int currentSheetViewHeight;
     private boolean hasIntercepted;
     private float peek;
+    private float elasticLimit;
 
-    /** Some values we need to manage width on tablets */
```
Comment (hzsweers) [inline bottomsheet/src/main/java/com/flipboard/bottomsheet/BottomSheetLayout.java]: Can you restore the comments to the way they were before?

File: bottomsheet/src/main/java/com/flipboard/bottomsheet/BottomSheetLayout.java [hunk 3]
```diff
@@ -517,6 +528,10 @@ public void onAnimationEnd(@NonNull Animator animation) {
      * Set the presented sheet to be in a peeked state.
      */
     public void peekSheet() {
+        if (getSheetView().getMeasuredHeight() < BottomSheetLayout.this.getMeasuredHeight()) {
```
Comment (hzsweers) [inline bottomsheet/src/main/java/com/flipboard/bottomsheet/BottomSheetLayout.java]: This logic change seems unrelated, what's the reasoning here?

Comment (hzsweers) [inline bottomsheet/src/main/java/com/flipboard/bottomsheet/BottomSheetLayout.java]: Ah wait, I see where you mentioned this now. The logic seems backward though, shouldn't it only expand if the measured height is equal to or greater than the layout?

Also, I think this would only apply in cases of wrap_content, which should be checked as well

**Comment by hzsweers** [comment]
Could you maybe include a gif/video displaying the behavior change?

**Comment by markrietveld** [comment]
Closing as inactive. We can come back to this later of course.
